Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -119,6 +119,10 @@ public final class SwiftLanguageServer: ToolchainLanguageServer {
 
   var commandsByFile: [DocumentURI: SwiftCompileCommand] = [:]
 
+  var documentParseTransition: [DocumentURI: (previousTree: SourceFileSyntax, collector: IncrementalParseNodeAffectRangeCollector)] = [:]
+  
+  public var documentReusedNodeCollector: [DocumentURI: IncrementalParseReusedNodeCollector] = [:]
+
```

Comment:
We shouldn’t be recording which nodes got re-used nodes unless it’s needed for testing purposes. I would suggest that we change `IncrementalParseReusedNodeCollector` from a delegate to a closure as I commented in https://github.com/apple/swift-syntax/pull/1685/files#r1254465986. Then this one could be an optional closure as well and you need to explicitly set it in the test cases to make it be called.

I would also add a comment that it’s for testing purposes only.

Reply:
I've added a `*For Testing*` comment there.

I will sync this design change here when I change the delegate to a closure in `SwiftParser`.